### PR TITLE
must-gather: collect noobaa logs independent of sc and external sc

### PR DESCRIPTION
**Pull request by rajatsing**
Signed-off-by: RAJAT SINGH <rajasing@redhat.com>

### CODE REVIEW

**Review by crombus — CHANGES_REQUESTED** [review]
We do not want to collect noobaa if stroage cluster is present.

You need different `if` section to do this with same condition storagecluster condition.

**Comment by rajatsing** [comment]
> We do not want to collect noobaa if stroage cluster is present.
> 
> You need different `if` section to do this with same condition storagecluster condition.

Noo, I think we need noobaa collection when storagecluster is present as well as when external storagecluter is present.

**Comment by jarrpa** [comment]
If I understand this correctly, the actual feature is to collect NooBaa information WITHOUT Ceph logs when you have an external StorageCluster, right? If so, a cleaner solution would be to introduce an `if !external` stanza in the existing `else` clause. I would imagine gathering NooBaa resources is independent of whether a StorageCluster exists or not, you should just be gathering any NooBaa resources that exist. This is especially important for our NooBaa-only deployments.

**Comment by crombus** [comment]
my concern regarding the storagecluster was because in last release we had a bug that noobaa shouldn’t be collected if storagecluster not present. Now as we can have noobaa without storagecluster also. So @rajatsing now no checks needed just call the script as we call for ns and clusterdscope scripts and update the log line w.r.this.

**Comment by rajatsing** [comment]
@crombus , I have made changes according to what you indicated. Now, noobaa is independent of storage and noobaa cluster

File: must-gather/collection-scripts/gather [hunk 1]
```diff
@@ -16,16 +16,18 @@ pre-install.sh
...
-   echo "Skipping the ceph and nooobaa collection as External Storage is enabled" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+externalStorageClusterPresent=$(oc get storagecluster -n openshift-storage -o go-template='{{range .items}}{{.spec.externalStorage.enable}}{{"\n"}}{{end}}')
+
+if [[ -n "${externalStorageClusterPresent}" ]]; then
+    echo "Skipping the ceph collection as External Storage is enabled" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
 elif [ -z "${storageClusterPresent}" ]; then
-    echo "Skipping ceph and nooobaa collection as Storage Cluster is not present" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+    echo "Skipping ceph collection as Storage Cluster is not present" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
 else
     echo "Collecting ceph and noobaa logs" | tee -a "${BASE_COLLECTION_PATH}"/gather-debug.log
```
Comment (crombus) [inline must-gather/collection-scripts/gather:29]: remove noobaa

**Review by crombus — CHANGES_REQUESTED** [review]
code changes looks good. Just update the log msg.

File: must-gather/collection-scripts/gather (hunk 1 above)
Comment (rajatsing) [reply must-gather/collection-scripts/gather:29]: Thanks, doing right now.

Comment (rajatsing) [reply must-gather/collection-scripts/gather]: done!.

**Review by crombus — APPROVED** [review]

**Pull request merged**